### ISSUE TITLE
[New Concept Exercise] : docstrings

### ISSUE THREAD

**Issue by aldraco** [exercism/python#2291]
This issue describes how to implement the `docstrings` concept exercise for the Python track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

This exercise should demonstrate and motivate the proper use of docstrings in Python, according to [PEP 257](https://www.python.org/dev/peps/pep-0257/).

## Learning objectives

* What is a docstring?
* `__doc__` dunder / magic attribute
* Student should implement a one-line docstring (e.g. for a function and/or method)
* Student should implement a multi-line docstring, properly formatted

Ideally, this exercise would have a pre-written module, which contains a class, which has several methods on it. The student should write the documentation for
* the module
* the class
* methods within the class

Since this is an automated exercise, there isn't a lot of room for discussion of the content of the docstrings, but we can check for its existence, formatting (a blank line between summary and paragraph, for example) and an approximate length that we feel is appropriate.

## Out of scope

 * Enforcing a specific docstring convention for styling (i.e. [reST](https://www.python.org/dev/peps/pep-0287/) vs [google's style guide](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#doc-function-args)). 

The examples should probably be in reST since it's fairly prevalent, but the formatting here is more related to a specific implementation of a documentation parser, not a generalized understanding of what docstrings are and how they should be used.

 * return types

Mentioning that return types can/should be documented is a good idea, but not necessary for the scope of this exercise, since it can also be covered more thoroughly by a `typing` exercise or similar.

## Concepts

- `docstrings`

## Prerequisites

- `comments`
 - `functions`
 - `methods`
 - `classes`
 - `modules`

## Resources to refer to

https://www.python.org/dev/peps/pep-0257/
https://www.python.org/dev/peps/pep-0287/

https://docs.python-guide.org/writing/documentation/#docstrings-and-magic
https://docs.python-guide.org/writing/documentation/#sphinx

### Hints

The following should be explicilty tested for:

 - a summary line
 - a blank line between the summary and the larger docstring
 - docstring should exist for the module, class, and methods

### After

Now is a good time to think about introducing the student to the magic (and usefulness) of automated documentation generators, or exposing them to prevalent examples of them in the Python community.

## Representer

?

## Analyzer

?

## Implementing

Tests should be written using `unittest.TestCase`, and the test file named `docstrings_test.py`.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by BethanyG** [comment]
Putting this on hold until we can get the issue updated with new specs.

**Comment by BethanyG** [comment]
I don't like you `stalebot`.  Not even a little.

**Comment by BethanyG** [comment]
This needs more detail and a write-up under the new issue template.  Closing for now until we can get to that.
